cpp/src/arrow/io/buffered.cc

```diff
@@ -311,7 +326,7 @@ class BufferedInputStream::Impl : public BufferedBase {
     }
 
     // Increase the buffer size if needed.
-    if (nbytes > buffer_->size() - buffer_pos_) {
+    if (nbytes > buffer_size_ - buffer_pos_) {
```

Comment:
Should the DCHECK below be changed as well?

Reply:
I'll resume `buffer_->size()` but I think they're same in `BufferedBase`